### ISSUE TITLE
Hello, When I use the dataset to train faster rcnn，I have set the filter_cfg=dict (filter_impty_gt=True, min_2=16) in the train_dataloader in the configuration file. I found that only over 800 images were loaded during training. Is this correct?

### ISSUE THREAD

**Issue by kNDSFLJ** [zcablii/SARDet_100K#59]
Hello, When I use the dataset to train faster rcnn，I have set the filter_cfg=dict (filter_impty_gt=True, min_2=16) in the train_dataloader in the configuration file. I found that only over 800 images were loaded during training. Is this correct?

This is my cocodetectability. py file
# dataset settings
dataset_type = 'CocoDataset'
data_root = 'data/coco/'

# Example to use different file client
# Method 1: simply set the data root and let the file I/O module
# automatically infer from prefix (not support LMDB and Memcache yet)

# data_root = 's3://openmmlab/datasets/detection/coco/'

# Method 2: Use `backend_args`, `file_client_args` in versions before 3.0.0rc6
# backend_args = dict(
#     backend='petrel',
#     path_mapping=dict({
#         './data/': 's3://openmmlab/datasets/detection/',
#         'data/': 's3://openmmlab/datasets/detection/'
#     }))
backend_args = None

train_pipeline = [
    dict(type='LoadImageFromFile', backend_args=backend_args),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', scale=(800, 800), keep_ratio=True),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PackDetInputs')
]
test_pipeline = [
    dict(type='LoadImageFromFile', backend_args=backend_args),
    dict(type='Resize', scale=(800, 800), keep_ratio=True),
    # If you don't have a gt annotation, delete the pipeline
    dict(type='LoadAnnotations', with_bbox=True),
    dict(
        type='PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                   'scale_factor'))
]
train_dataloader = dict(
    batch_size=2,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    batch_sampler=dict(type='AspectRatioBatchSampler'),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='Annotations/train.json',
        data_prefix=dict(img='Images/train'),
        filter_cfg=dict(filter_empty_gt=True, min_size=16),
        pipeline=train_pipeline,
        backend_args=backend_args))
val_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='Annotations/val.json',
        data_prefix=dict(img='Images/val'),
        test_mode=True,
        pipeline=test_pipeline,
        backend_args=backend_args))
test_dataloader = val_dataloader

val_evaluator = dict(
    type='CocoMetric',
    ann_file=data_root + 'Annotations/val.json',
    metric='bbox',
    format_only=False,
    backend_args=backend_args)

test_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='Annotations/test.json',
        data_prefix=dict(img='Images/test'),
        test_mode=True,
        pipeline=test_pipeline,
        backend_args=backend_args))

test_evaluator = dict(
    type='CocoMetric',
    ann_file=data_root + 'Annotations/test.json',
    metric='bbox',
    classwise = True,
    format_only=False,
    backend_args=backend_args)

**This is the log from the beginning of my training**
05/07 22:16:21 - mmengine - INFO - load model from: torchvision://resnet50
05/07 22:16:21 - mmengine - INFO - Loads checkpoint by torchvision backend from path: torchvision://resnet50
05/07 22:16:22 - mmengine - WARNING - The model and loaded state dict do not match exactly

unexpected key in source state_dict: fc.weight, fc.bias

05/07 22:16:22 - mmengine - WARNING - "FileClient" will be deprecated in future. Please use io functions in https://mmengine.readthedocs.io/en/latest/api/fileio.html#file-io
05/07 22:16:22 - mmengine - WARNING - "HardDiskBackend" is the alias of "LocalBackend" and the former will be deprecated in future.
05/07 22:16:22 - mmengine - INFO - Checkpoints will be saved to F:\jzh\mmdetection-main\run_workstation.
05/07 22:16:36 - mmengine - INFO - Epoch(train)  [1][ 50/419]  lr: 2.0000e-05  eta: 0:23:50  time: 0.2874  data_time: 0.1239  memory: 1217  loss: 1.6669  loss_rpn_cls: 0.6901  loss_rpn_bbox: 0.1042  loss_cls: 0.7730  acc: 95.4102  loss_bbox: 0.0996

**Comment by zcablii** [comment]
How many GPUs do you use?
The train set has 94,493 images. Therefore, total iterations shown in the progress log should be: ⌈94,493 / (batch_size * GPUs)⌉

**Comment by kNDSFLJ** [comment]
I use a 4090d.

**Comment by kNDSFLJ** [comment]
> 您使用多少个 GPU？火车集有 94,493 张图像。因此，进度日志中显示的总迭代次数应为：⌈94,493 / （batch_size * GPU）⌉
I use a 4090d.

**Comment by zcablii** [comment]
Sorry, I accidentally deleted your reply. I saw you said that you used one GPU. The train set has 94,493 images. Therefore, total iterations shown in the progress log should be: ⌈94,493 / (batch_size * GPUs)⌉
You may need to check the dataset. @kNDSFLJ
